networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -91,10 +91,9 @@ public Optional<RpcException> validateRequest(
     final int maxRequestBlobSidecars = specConfig.getMaxRequestBlobSidecars();
     final int maxBlobsPerBlock = specConfig.getMaxBlobsPerBlock();
 
-    final long requestedCount = calculateRequestedCount(request, maxBlobsPerBlock);
+    final int requestedCount = calculateRequestedCount(request, maxBlobsPerBlock);
```

Comment:
NIT: I think we don't need another method just to handle the exception and return -1. It kinda hides that we are using -1 as a "flag" for ArithimeticException.

Here is a suggestion (and we can remove the `calculateRequestedCount` method.

```suggestion
    int requestedCount;
    try {
      requestedCount = request.getCount().times(maxBlobsPerBlock).intValue();
    } catch (final ArithmeticException e) {
      requestedCount = -1;
    }
```

Reply:
Implemented this. I like it better. I didn't remove the method since it is used in `onIncomingMessage` but it will never fail there, since the message would have been rejected anyways.